src/components/atoms/dropdown.tsx

```diff
@@ -0,0 +1,96 @@
+import { CaretDown, CaretUp, X } from '@phosphor-icons/react'
+import { useState } from 'react'
+
+interface DropdownProps {
+  readonly options: string[]
+  readonly label: string
+  readonly onOptionSelect: (option: string | null) => void
+}
+
+export default function Dropdown({ options, label, onOptionSelect }: DropdownProps) {
+  const [searchTerm, setSearchTerm] = useState<string>('')
+  const [isOpen, setIsOpen] = useState<boolean>(false)
+  const [selectedOption, setSelectedOption] = useState<string | null>(null)
+
+  const filteredOptions = options.filter(option =>
+    option.toLowerCase().includes(searchTerm.toLowerCase()),
+  )
+
+  function handleOptionClick(option: string) {
+    console.log('Selected option:', option)
```

Comment:
Is this for debugging? If yes, don't forget to remove it!

Reply:
CC: @bryannaufal